### PR TITLE
ci: use all cpu cores

### PR DESCRIPTION
**Pull request by amiclaus**
## Pull Request Description

Add NUM_JOBS variable holding the available number of cpu cores on a build agent.

Use all avaliable cores for all ci builds.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [ ] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [ ] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [x] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: ci/run_build.sh [hunk 1]
```diff
@@ -39,6 +39,7 @@ sudo apt-get update
 
 TOP_DIR="$(pwd)"
 DEPS_DIR="${TOP_DIR}/deps"
+NUM_JOBS=$(nproc)
```
Comment (rbolboac) [inline ci/run_build.sh:42]: just wondering if this won't freeze the system in some cases

**Review by SRaus — CHANGES_REQUESTED** [review]
I don't think it's ok to use all cores for building projects. 
The No-OS agent is also used by small jobs from other Azure pipelines where internal agents are required (like uploading RPI boot files on Artifactory server etc). But even if would dedicated only for No-OS, there is required at least one core for pipeline and other background processes.
So please change "make -j${NUM_JOBS -1}" or  "NUM_JOBS=$(nproc -1)".

**Comment by amiclaus** [comment]
> I don't think it's ok to use all cores for building projects. The No-OS agent is also used by small jobs from other Azure pipelines where internal agents are required (like uploading RPI boot files on Artifactory server etc). But even if would dedicated only for No-OS, there is required at least one core for pipeline and other background processes. So please change "make -j${NUM_JOBS -1}" or "NUM_JOBS=$(nproc -1)".

This has nothing to do with the no-OS agent.

If you take a look in the code, these scripts are run only on azure VMs which are handled **strictly** by https://github.com/analogdevicesinc/no-OS/blob/master/azure-pipelines.yml

File: ci/run_build.sh (hunk 1 above)
Comment (amiclaus) [reply ci/run_build.sh:42]: Hmm, you mean calling the `nproc` command? or using all cores for azure VMs which btw were already forced in some cases with hardcoded values (see for example: https://github.com/analogdevicesinc/no-OS/blob/bc77edd35f9f42a9fb5289e21994dca8fdad7bbe/ci/astyle.sh#L99) ?

**Comment by SRaus** [comment]
> > I don't think it's ok to use all cores for building projects. The No-OS agent is also used by small jobs from other Azure pipelines where internal agents are required (like uploading RPI boot files on Artifactory server etc). But even if would dedicated only for No-OS, there is required at least one core for pipeline and other background processes. So please change "make -j${NUM_JOBS -1}" or "NUM_JOBS=$(nproc -1)".
> 
> This has nothing to do with the no-OS agent.
> 
> If you take a look in the code, these scripts are run only on azure VMs which are handled **strictly** by https://github.com/analogdevicesinc/no-OS/blob/master/azure-pipelines.yml

Even so, aren't there any background processes - on "ubuntu-latest" container - for which we'll need to keep one CPU core?

File: ci/run_build.sh (hunk 1 above)
Comment (rbolboac) [reply ci/run_build.sh:42]: I mean using all cores. but if we already do it in some cases, I guess there is no issue

**Review by SRaus — APPROVED** [review]

**Review by rbolboac — APPROVED** [review]

**Pull request merged**